src/FSharp.Data.GraphQL.Server.Middlewares.AspNetCore/MultipartRequest.fs

```diff
@@ -0,0 +1,99 @@
+namespace FSharp.Data.GraphQL.Server.Middlewares.AspNetCore
+
+open Microsoft.AspNetCore.WebUtilities
+open Newtonsoft.Json
+open System.Collections.Generic
+open Newtonsoft.Json.Linq
+
+/// A GraphQL operation request.
+type Operation =
+      /// Contains the query used in this operation.
+    { Query : string
+      /// Contains variables used by this operation.
+      Variables : Map<string, obj> }
+
+/// <summary> A GrahpQL request using multipart request specification. </summary>
+/// <remarks> For more information, see https://github.com/jaydenseric/graphql-multipart-request-spec. </remarks>
+type MultipartRequest =
+      /// Contains the list of operations of this request.
+      /// If the request is not batched, then the single operation will be inside this list as a singleton.
+    { Operations : Operation list }
+
+/// Contains tools for working with GraphQL multipart requests.
+module MultipartRequest =
+    let private parseOperations (operations: string) (map : IDictionary<string, string list>) (files : IDictionary<string, File>) : Operation list =
+        let operations =
+            match JToken.Parse(operations) with
+            | :? JArray as ops -> ops.ToObject<Operation list>()
+            | :? JObject as op -> [ op.ToObject<Operation>() ]
+            | _ -> failwith "Unexpected operations value."
+        let map = 
+            map 
+            |> Seq.map (fun x -> x.Key, x.Value |> Seq.tryHead)
+            |> Seq.filter (fun (_, v) -> Option.isSome v)
+            |> Seq.map (fun (k, v) -> k, v.Value)
+            |> Map.ofSeq
+        let files =
+            files
+            |> Seq.map (fun x -> x.Key, x.Value)
```

Comment:
You can use `KeyValue` acive pattern: `Seq.map (|KeyValue|)`.

Reply:
Done.